src/components/screens/forms/form-replies-screen.tsx

```diff
@@ -0,0 +1,278 @@
+
+    const gridColumns = managementListColumns.map<GridColDef>(column => ({
+      field: column.name || "",
+      headerName: column.title,
+      allowProps: true,
+      flex: 1,
+      renderHeader: params => {
+        return (
+          <AdminFormRepliesScreenStack direction="row">
+            <AdminFormRepliesScreenText sx={{ fontWeight: "bold" }}>{ params.colDef.headerName }</AdminFormRepliesScreenText>
+          </AdminFormRepliesScreenStack>
+        );
+      },
+      renderCell: params => {
+        return (
+          <AdminFormRepliesScreenStack direction="row">
+            <AdminFormRepliesScreenText>{ column.name ? params.row[column.name] : "" }</AdminFormRepliesScreenText>
+          </AdminFormRepliesScreenStack>
+  
+        );
+      }
+    }));
+
+    if (keycloak?.hasRealmRole(ApplicationRoles.METAFORM_ADMIN)) {
+      gridColumns.push({
+        field: "actions",
+        type: "actions",
+        width: 80,
+        getActions: (params: { row: any; }) => [
+          <GridActionsCellItem
+            icon={ <DeleteIcon/> }
+            onClick={ () => setDeletableReplyId(params.row.replyId) }
+            label={ strings.generic.delete }
+            showInMenu
+          />
+        ]
+      } as GridColDef);
+    }
+    
+    return gridColumns;
+  };
+
+  /**
+   * Deletes a reply
+   * 
+   * @param replyId reply id
+   */
+  const deleteReply = async (replyId: string) => {
+    setLoading(true);
+
+    try {
+      if (!metaform || !metaform.id || !replyId) {
+        return;
+      }
+    
+      await repliesApi.deleteReply({
+        metaformId: metaform.id,
+        replyId: replyId
+      });
+  
+      setRows(rows?.filter(row => row.replyId !== replyId));
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.adminRepliesScreen.deleteReply, e);
+    }
+
+    setLoading(false);
+  };
+  
+  /**
+   * Event handler for delete reply dialog confirm
+   */
+  const onReplyDeleteConfirm = async () => {
+    if (deletableReplyId) {
+      deleteReply(deletableReplyId);
+    }
+  
+    setDeletableReplyId(undefined);
+  };
+
+  /**
+   * Replies screen setup
+   */
+  const setup = async () => {
+    setLoading(true);
+
+    try {
+      const metaformData = await metaformsApi.findMetaform({ metaformId: formId! });
+      const repliesData = await repliesApi.listReplies({ metaformId: formId! });
+      setMetaform(metaformData);
+      const fields = await getManagementListFields(metaformData);
+      if (repliesData && fields) {
+        const replyRows = repliesData.map(reply => (buildRow(reply, fields)));
+        if (showAllReplies) {
+          setRows(replyRows);
+        } else {
+          setRows(replyRows.filter(row => row.status === ReplyStatus.WAITING));
+        }
+      }
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.adminRepliesScreen.fetchReplies, e);
+    }
+  
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setup();
+  }, [showAllReplies]);
```

Comment:
Is it really necessary to reaload everything when showAllReplies changes? Couldn't we instead use filtered version of rows?